File: Programmars/Level_2/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    word = ''
    for letter in s:
        if letter == ' ':
            answer += word + letter
            word = ''
        else:
            if word:
                word += letter.lower()
            else:
                word += letter.upper()
    if word:
        answer += word
    return answer


print(solution("3people unFollowed me"))  # "3people Unfollowed Me"
print(solution("for the last week"))  # "For The Last Week"
print(solution("abc"))
print(solution(" adgagd 3eagdag "))


# ref: join & capitalize & split 활용
def solution(s):
    return ' '.join([word.capitalize() for word in s.split(" ")])

File: JavaScript/Programmars/Level_2/JadenCase 문자열 만들기.js
// 문제: https://programmers.co.kr/learn/courses/30/lessons/12951
function solution(s) {
  let answer = '';

  let tempWord = '';
  for (let char of s) {
    if (char === ' ') {
      if (tempWord.length > 0) {
        answer += tempWord[0].toUpperCase() + tempWord.substr(1).toLowerCase();
        tempWord = '';
      }
      answer += ' ';
    } else tempWord += char;
  }

  if (tempWord.length > 0)
    answer += tempWord[0].toUpperCase() + tempWord.substr(1).toLowerCase();

  return answer;
}

// 다른 풀이: ✅ charAt() => 대상 문자열이 공백인 경우 빈 문자열 리턴 ([index]는 undefined 리턴)
// 참고: https://thisthat.dev/string-char-at-vs-string-bracket-notation/
function solution(s) {
  return s
    .split(' ')
    .map((v) => v.charAt(0).toUpperCase() + v.substring(1).toLowerCase())
    .join(' ');
}

// 250417
function solution(s) {
  return s
    .split(' ')
    .map((word) => {
      if (!word) return '';
      return word[0].toUpperCase() + word.slice(1).toLowerCase();
    })
    .join(' ');
}

console.log(solution('3people unFollowed me')); // "3people Unfollowed Me"
console.log(solution('for the last week')); // "For The Last Week"
